pkg/sql/app_stats.go

```diff
@@ -113,6 +115,10 @@ func (a *appStats) recordStatement(
 	// Get the statistics object.
 	s := a.getStatsForStmt(stmtKey{stmt: buf.String(), failed: err != nil, distSQLUsed: distSQLUsed})
 
+	// Make sure to add statement recording to the overhead latency! The above
+	// FormatNode call is particularly expensive.
+	ovhLat += timeutil.Now().Sub(now).Seconds()
```

Comment:
Note that maybe there's a case to be made for removing `recordStatementSummary` altogether and merging its functionality in `recordStatement`.